File: ecos-records/src/main/java/ru/citeck/ecos/records3/graphql/meta/value/factory/BooleanValueFactory.java
package ru.citeck.ecos.records3.graphql.meta.value.factory;

import ru.citeck.ecos.records3.graphql.meta.value.MetaValue;

import java.util.Collections;
import java.util.List;

public class BooleanValueFactory implements MetaValueFactory<Boolean> {

    @Override
    public MetaValue getValue(Boolean value) {
        return new MetaValue() {
            @Override
            public String getString() {
                return value.toString();
            }

            @Override
            public Boolean getBool() {
                return value;
            }
        };
    }

    @Override
    public List<Class<? extends Boolean>> getValueTypes() {
        return Collections.singletonList(Boolean.class);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/factory/BooleanValueFactory.kt
package ru.citeck.ecos.records3.record.atts.value.factory

import ru.citeck.ecos.records3.record.atts.value.AttValue

class BooleanValueFactory : AttValueFactory<Boolean> {

    override fun getValue(value: Boolean): AttValue? {

        return object : AttValue {

            override fun asText(): String {
                return value.toString()
            }

            override fun asBoolean(): Boolean {
                return value
            }

            override fun asRaw(): Any {
                return value
            }
        }
    }

    override fun getValueTypes() = listOf(java.lang.Boolean::class.java)
}
